sklearn/datasets/kddcup99.py

```diff
@@ -11,32 +11,35 @@
 import sys
 import errno
 from gzip import GzipFile
-from io import BytesIO
 import logging
 import os
 from os.path import exists, join
-try:
-    from urllib2 import urlopen
-except ImportError:
-    from urllib.request import urlopen
 
 import numpy as np
 
+
+from .base import _fetch_url
 from .base import get_data_home
+from .base import RemoteFileMetadata
 from ..utils import Bunch
 from ..externals import joblib, six
 from ..utils import check_random_state
 from ..utils import shuffle as shuffle_method
 
 
-URL10 = ('http://archive.ics.uci.edu/ml/'
-         'machine-learning-databases/kddcup99-mld/kddcup.data_10_percent.gz')
-
-URL = ('http://archive.ics.uci.edu/ml/'
-       'machine-learning-databases/kddcup99-mld/kddcup.data.gz')
+ARCHIVE = RemoteFileMetadata(
```

Comment:
I would rather have one comment per (and close to the) RemoteFileMetadata object so that it is competely straightfoward to figure out the mapping from figshare URLs to the original URLs.